vaadin-messages-flow-parent/vaadin-messages-flow-integration-tests/src/test/java/com/vaadin/flow/component/messages/tests/MessageInputIT.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2000-2021 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.messages.tests;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
+
+import com.vaadin.flow.component.html.testbench.InputTextElement;
+import com.vaadin.flow.component.messages.testbench.MessageInputElement;
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.tests.AbstractComponentIT;
+
+@TestPath("vaadin-messages/message-input-test")
+public class MessageInputIT extends AbstractComponentIT {
+
+    private MessageInputElement messageInput;
+
+    @Before
+    public void init() {
+        open();
+        messageInput = $(MessageInputElement.class).first();
+    }
+
+    @Test
+    public void submitValue_eventHasCorrectValue() {
+        getInShadowRoot(messageInput, By.tagName("vaadin-text-area"))
+                .sendKeys("foo");
+        getInShadowRoot(messageInput, By.tagName("vaadin-button")).click();
```

Comment:
I recommend adding `submit` method to `MessageInputElement`.

Also, `getInShadowRoot` is an old util method that is no longer needed. The recommended way is to use the TestBench query method `$`, which knows to find children also from the shadow root. So the following should work:
```java
messageInput.$("vaadin-button");
```

Reply:
Oh, I've just had the same idea in the above comment 😄 What about a `submit` method with a string parameter, then?